core/domain/exp_jobs_one_off_test.py

```diff
@@ -1241,8 +1241,8 @@ def test_exp_state_pairs_are_produced_only_for_desired_interactions(self):
             exp_jobs_one_off.ItemSelectionInteractionOneOffJob.get_output(
                 job_id))
         expected_output = [(
-            '[u\'exp_id0\', '
-            '[u\'State2: <p>This is value3 for ItemSelection</p>\']]'
+            u'[u\'exp_id0\', '
```

Comment:
I am not sure why the comparison of strings (unicode string and simple string) fails here even though it works fine in other test. Can you please look into this, Thanks!

Reply:
Looking at the logs, I think the list is ordered differently, and hence the output strings don't match.

Reply:
The output differs based on which exploration is checked first. So, I have added two cases of expected output and tested if actual output matches any of those. PTAL. Thanks!

Reply:
Hmmm.... This is weird. Typically in tests you should be able to define exactly what the assertions are. Maybe, something along the lines of does output contain str1 and str2 would be better? Thinking about it, what we want is that the array must contain an entry corresponding to both the states (irrespective or order). So that seems to be the right thing to do. 

Basically, don't have 2 asserts in a try catch. Instead assert that the string contains the desired entries.

Reply:
Would it be fine to do something along these lines:
```
        self.assertEqual(len(actual_output), 2)
        self.assertEqual(actual_output[0], '[u\'2\', [u\'exp_id0 State1\']]')

        is_second_string_in_output_correct = False
        if (
                actual_output[1] == (
                    '[u\'1\', [u\'exp_id0 State2\', u\'exp_id1 State1\']]') or
                actual_output[1] == (
                    '[u\'1\', [u\'exp_id1 State1\', u\'exp_id0 State2\']]')
            ):
            is_second_string_in_output_correct = True
        self.assertEqual(is_second_string_in_output_correct, True)
```

Reply:
Yes, that seems good. I have made the changes accordingly, PTAL. Thanks!

Reply:
I don't see this change? Did you finally hardcode a string?

Reply:
No, I did not hardcode the strings. You can see the change [here](https://github.com/oppia/oppia/pull/5972/commits/e119cc39b0c4d58cbb50fffb10912e4fa3d9966e#diff-ceaab7e54d8ce7ddffb0f2c94f455c9eR1146).

Reply:
Oh, was this comment at the wrong line? Thanks for the link to the line :)

Reply:
Ah no, lines actually changed due to addition of code after a review by @bansalnitish.

Reply:
Oh I see, never mind. I was confused about how nothing changed at the final assertion line. read the code again, and I think it looks reasonable! Thanks :)

Reply:
You're welcome!